### PR TITLE
Backport fix for #8080

### PR DESCRIPTION
**Pull request by jasonwbarnett**
### Description

Backport fix for #8080 to Chef 14.

### Issues Resolved

- #8080 

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] RELEASE\_NOTES.md has been updated if required (not required for bugfixes, required for API changes)
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by jasonwbarnett** [comment]
takes care of backporting #8278 into chef 14.

**Comment by btm** [comment]
@jasonwbarnett thanks. did you see this in the appveyor unit test failures?
```
1) Chef::Provider::Service::Windows load_current_resource Chef::Provider::Service::Windows start_service run_as_user user is specified configures service run_as_user and run_as_password
     Failure/Error: Win32::Service.start(@new_resource.service_name)
     NoMethodError:
       undefined method `start' for Win32::Service:Class
     # ./lib/chef/provider/service/windows.rb:101:in `block in start_service'
     # ./lib/chef/provider/service/windows.rb:341:in `block in spawn_command_thread'
  2) Chef::Provider::Service::Windows load_current_resource Chef::Provider::Service::Windows start_service run_as_user user is not specified does not configure service run_as_user and run_as_password
     Failure/Error: Win32::Service.start(@new_resource.service_name)
     NoMethodError:
       undefined method `start' for Win32::Service:Class
     # ./lib/chef/provider/service/windows.rb:101:in `block in start_service'
     # ./lib/chef/provider/service/windows.rb:341:in `block in spawn_command_thread'
```

**Comment by jasonwbarnett** [comment]
ugh... I need to backport some of the refactored tests I did... love it.

File: spec/unit/provider/service/windows_spec.rb [hunk 1]
```diff
@@ -96,6 +96,7 @@
     Win32::Service::DEMAND_START = 0x00000003
     Win32::Service::DISABLED = 0x00000004
 
+    allow(Win32::Service).to receive(:start).with(any_args).and_return(Win32::Service)
```
Comment (jasonwbarnett) [inline spec/unit/provider/service/windows_spec.rb]: @btm this should fix the failing appveyor tests... only time will tell.

**Pull request merged**